server/src/main/java/org/elasticsearch/transport/InboundDecoder.java

```diff
@@ -38,23 +38,31 @@ public InboundDecoder(Version version, Recycler<BytesRef> recycler) {
         this.recycler = recycler;
     }
 
-    public int decode(ReleasableBytesReference reference, Consumer<Object> fragmentConsumer) throws IOException {
+    public int decode(
+        TcpChannel channel,
+        ReleasableBytesReference reference,
+        CheckedBiConsumer<TcpChannel, Object, IOException> fragmentConsumer
```

Comment:
Could we not pass the `TcpChannel` around everywhere like this? It's always the same channel isn't it?

Reply:
Probably, I bet we can make additional optimizations like that (though I might actually like it better to be less "functional" here and just remove the ability to pass a consumer here when we only have one production code user of this method). It's weird to have this kind of generality on the very very hot path (though admittedly it's not showing up in profiling) just for tests?

Reply:
Hiding my above comment, I misread the question initially. I added passing around the channel here to avoid instantiating a capturing lambda here where previously we had a non-capturing one.